src/main/java/com/synopsys/integration/alert/web/config/ConfigActions.java

```diff
@@ -165,8 +179,20 @@ public String testConfig(final FieldModel restModel, final String destination) t
         if (descriptorActionApi.isPresent()) {
             final DescriptorActionApi descriptorApi = descriptorActionApi.get();
             final TestConfigModel testConfig = descriptorApi.createTestConfigModel(restModel, destination);
-            final List<ConfigField> configFields = retrieveUIConfigFields(testConfig.getFieldModel());
-            descriptorApi.testConfig(configFields, testConfig);
+            final Map<String, ConfigField> fieldsToTest = new LinkedHashMap<>();
+            final Map<String, ConfigField> configFields = retrieveUIConfigFields(testConfig.getFieldModel())
+                                                              .stream()
+                                                              .collect(Collectors.toMap(ConfigField::getKey, Function.identity()));
+            final ConfigContextEnum descriptorContext = EnumUtils.getEnum(ConfigContextEnum.class, restModel.getContext());
+
+            if (ConfigContextEnum.DISTRIBUTION == descriptorContext) {
+                final Map<String, ConfigField> globalFields = createConfigFields(descriptorContext, restModel.getDescriptorName()).stream()
+                                                                  .collect(Collectors.toMap(ConfigField::getKey, Function.identity()));
+                fieldsToTest.putAll(globalFields);
+                fieldsToTest.putAll(includeProviderFields(descriptorContext, restModel));
+            }
+            fieldsToTest.putAll(configFields);
```

Comment:
We should convert this map to FieldAccessor before passing it to the actionApi. All the child classes that define this method immediately convert it into FieldAccessor.